.github/workflows/functions-deploy.yml

```diff
@@ -12,6 +12,8 @@ on:
 jobs:
   deploy-functions:
     runs-on: ubuntu-latest
+    environment: 
+      name: '${{ env.FIREBASE_PROJECT }}'
```

Comment:
I've added `production` and `develop` Github environments with Firebase service account secrets for the corresponding projects